modules/javafx.graphics/src/main/java/com/sun/javafx/application/preferences/PlatformPreferences.java

```diff
@@ -76,8 +75,7 @@ public final class PlatformPreferences extends AbstractMap<String, Object> imple
     /** Contains the implementation of the property-based API. */
     private final PreferenceProperties properties = new PreferenceProperties(this);
 
-    private final List<InvalidationListener> invalidationListeners = new CopyOnWriteArrayList<>();
-    private final List<MapChangeListener<? super String, Object>> mapChangeListeners = new CopyOnWriteArrayList<>();
+    private MapListenerHelper<String, Object> helper;
```

Comment:
I wonder if this change should be undertaken as a separate PR, it's a totally unrelated functionality.